### ISSUE TITLE
Integration with rematch

### ISSUE THREAD

**Issue by theunreal** [prescottprue/react-redux-firebase#904]
**Do you want to request a *feature* or report a *bug*?**  

Bug or documentation request, not sure.

(If this is a *usage question*, please **do not post it here**—post it on [gitter](https://gitter.im/redux-firebase/Lobby). If this is not a “feature” or a “bug”, or the phrase “How do I...?” applies, then it's probably a usage question.)

**What is the current behavior?**
App fails to load, error: 
`Invariant Violation: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:`

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [codesandbox](https://codesandbox.io/) or similar.**

Code:

```
// Create plugins
const persistPlugin = createPersistPlugin({
  version: 2,
  storage: AsyncStorage,
  blacklist: [],
});
const loadingPlugin = createLoadingPlugin({});


const createStoreWithFirebase = compose(
  ReactReduxFirebaseProvider(firebase,
    {
      userProfile: 'users',
      useFirestoreForProfile: true,
      enableLogging: true,
    }),
  reduxFirestore(firebase),
)(createStore);

const configureStore = () => {
  const store = init({
    models,
    redux: {
      createStore: createStoreWithFirebase,
      middlewares: [],
      reducers: {
        firebase: firebaseReducer,
        firestore: firestoreReducer,
      },
    },
    plugins: [persistPlugin, loadingPlugin],
  });

  const persistor = getPersistor();
  const { dispatch } = store;

  return { persistor, store, dispatch };
};

export default configureStore;
```

**What is the expected behavior?**
Should work

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**
"@rematch/core": "^1.4.0",
"react-redux-firebase": "^3.3.0",
    "redux": "^4.0.5",
    "redux-firestore": "^0.13.0",

**Comment by prescottprue** [comment]
I don't believe that the way you are using the ReactReduxFirebaseProvider will work as expected. It seems like you are mixing the old API (which used a store enhancer named `reactReduxFirebase`( with the new API which has the ReactReduxFirebaseProvider (for using context)

**Comment by theunreal** [comment]
@prescottprue I really can't figure out how to put it both together. No answers at stackoverflow either

**Comment by younes0** [comment]
@prescottprue is the createStoreWithFirebase still needed ? 
react-redux-firebase doc doesn't mention it unlike redux-firestore doc

BTW, I'm also facing issues while integrating with easy-peasy. 
easy-peasy and rematch beeing very similar (and the most popular libs for easing redux usage)
https://github.com/ctrlplusb/easy-peasy/issues/524
